### ISSUE TITLE
Edge ticks sometimes disappear

### ISSUE THREAD

**Issue by EdwinChan** [matplotlib/matplotlib#6641]
I am using
- matplotlib 1.5.1 installed with pip for Python 3.5.1+ on Ubuntu; and
- matplotlib 2.0.0b1+1708.g5de9c44 built from source for Python 3.5.1 on Scientific Linux 5.8.

Consider the example
```python
import matplotlib
from matplotlib import pyplot

pyplot.xlim(0, 1.2)
pyplot.ylim(0, 1.2)
pyplot.annotate(matplotlib.__version__, (0, 0), (0.1, 1.1), 'data', 'data',
  ha='left', va='top')
pyplot.savefig('lim.png')
```
With the GitHub version of matplotlib, the last tick (1.2) is missing from both axes. This is a regression because matplotlib 1.5.1 can show both ticks without problem.
![lim](https://cloud.githubusercontent.com/assets/906137/16351207/f43b5324-3a31-11e6-88f1-47ff5ec5bc15.png)

**Comment by anntzer** [comment]
Bisected back to #5956.  I wonder if the check for `interval_contains` is really necessary?  Can't we just rely on the ticker to only return ticks that are within the interval?

**Comment by efiring** [comment]
It looks like that check dates back to 2005--tickers have changed a lot since then, so yes, it seems reasonable to leave the responsibility for tick selection entirely to the tickers.  It would be a good idea to check all of them to see whether they are handling floating point slop in a reasonable and consistent way.

**Comment by anntzer** [comment]
The log-ticker seems to use a strict inequality check (e.g. there's no tick at 10 for `xlim(1, nextafter(10, 0))`) but that's not really a problem (or at most a separate one), right?
Also I am now not even certain that the whole dance with `Base` to handle floating point errors is necessary?

**Comment by efiring** [comment]
The Base machinery was put in explicitly for that purpose, because we were having problems, I believe; is there another mechanism that replaces it?  (I didn't write Base, and haven't looked at it carefully.)

**Comment by efiring** [comment]
#5956 has an obvious effect only on tick *labels*, not on whether a tick is drawn, so I don't see how it could be the cause of the change in behavior.

**Comment by efiring** [comment]
Closed by #7042.
